### PR TITLE
Extend plaidbench batchnorm folding

### PR DESCRIPTION
**Pull request by tzerrell**
Expand the plaidbench network-level batchnorm folding so as to produce relevant perf metrics for EDSL-based PlaidML while waiting for the relevant constant folding to be fully implemented.

### CODE REVIEW

File: plaidbench/plaidbench/frontend_keras.py [hunk 1]
```diff
@@ -68,17 +68,23 @@ def setup_imdb(train, epoch_size):
...
             from keras.backend import set_learning_phase
             set_learning_phase(learn_phase)
         self.frontend = frontend
         self.params = params
+        self.fold_bn_for_backend = fold_bn_for_backend
 
-    def fold_batch_norm(self, model):
+    def fold_batch_norm(self, model, backend):
         import json
-        import plaidml.keras.backend as K
```
Comment (flaub) [inline plaidbench/plaidbench/frontend_keras.py]: AFAICT, we only ever use `K.get_value()` and `K.set_value()`, so why can't we always just use `import keras.backend as K`? The assumption is that `import keras.backend` will route to the appropriate backend.

File: plaidbench/plaidbench/frontend_keras.py [hunk 2]
```diff
@@ -343,7 +349,11 @@ def __init__(self, backend, fp16, train):
     def model(self, params):
         if self.train:
             return TrainingModel(self, params)
-        return InferenceModel(self, params)
+        if os.getenv('USE_STRIPE', '0') == '1' or self.backend == 'plaid_edsl':
```
Comment (flaub) [inline plaidbench/plaidbench/frontend_keras.py]: It seems like this should really be: 
```
enable_bn_folding = (os.getenv('USE_STRIPE', '0') == '1') or (self.backend == 'plaid_edsl')
return InferenceModel(self, params, enable_bn_folding=enable_bn_folding)
```

File: plaidbench/plaidbench/frontend_keras.py (hunk 1 above)
Comment (tzerrell) [reply plaidbench/plaidbench/frontend_keras.py]: `import keras.backend` does not actually route to the appropriate backend in the context of plaidbench in many circumstances due to technical debt lingering from when we had to monkey patch PlaidML into Keras. That said, it might be worthwhile to unwind that now rather than committing more hacky code like this

Comment (flaub) [reply plaidbench/plaidbench/frontend_keras.py]: At least in the context that this matters in (i.e. only the `plaid` and `plaid-edsl` backends), it appears to work properly (at least in the testing I did locally).

Comment (tzerrell) [reply plaidbench/plaidbench/frontend_keras.py]: hrm, it's been breaking my tests. I'm drilling into why though and it seems mostly unrelated, so I'm hoping I can clean this up.

Comment (tzerrell) [reply plaidbench/plaidbench/frontend_keras.py]: A major part of the broken code was adjusted in #527, so if you're not super up-to-date that might explain the discrepancy in our results
